### PR TITLE
Fixed kebab casing of previous `appwindow.slint`.

### PR DESCRIPTION
**Pull request by Enyium**
See <https://github.com/slint-ui/slint-rust-template/pull/22>.

### CODE REVIEW

**Comment by hunger** [comment]
We have way more "appwindow" (5) in file names in the slint repo than we have "app-window" (none).

But you are right: In slint files I would probably write app-window.slint... but not in rust/c++/python files.

I am undecided, sorry.

**Comment by Enyium** [comment]
> We have way more "appwindow" (5) in file names in the slint repo than we have "app-window" (none).

I don't understand what you're up to. With this PR, you will have `app-window` in filenames.
 
> But you are right: In slint files I would probably write app-window.slint... but not in rust/c++/python files.

What do you mean not wanting to write `app-window.slint`? If, e.g., the file is loaded from a Node.js context, you'll have to reference it by its name. It's something from a different context. The other context doesn't need preemptive obedience regarding the casing of where it'll be used.

Currently, with `appwindow`, you have flat case, which is also not how you write in Rust/C++/Python.

Casings may appear in contexts where they are foreign. E.g., Rust uses snake case for functions. But when you use the [`windows`](https://crates.io/crates/windows) crate, the functions of this foreign context are all in Pascal case. It's like a human being with a foreign accent.

But it isn't as if this change here would grow rampant. Regarding Python, e.g., it is planned to allow for underscores, even if the filenames use hyphens, which is parallel to the Cargo context with crate names and module names where case is converted.

File: api/python/README.md [hunk 1]
```diff
@@ -75,8 +75,9 @@ export component AppWindow inherits Window {
 ```python
 import slint
 
-# slint.loader will look in `sys.path` for `appwindow.slint`.
-class App(slint.loader.appwindow.AppWindow):
+# slint.loader will look in `sys.path` for `app-window.slint`.
+# class App(slint.loader.app_window.AppWindow):  # Supported soon.
+class App(getattr(slint.loader, "app-window").AppWindow):
```
Comment (tronical) [inline api/python/README.md:80]: I'd rather not have this in the release but implement #6216 first.

**Review by tronical — COMMENTED** [review]
Change looks good and completes this transition. Thanks :). I'll still hold off until after the release then and will merge and finish up.

**Comment by Enyium** [comment]
> I'll still hold off until after the release

Ah, so not until right before the release of v1.9. Then, this is not so bad. 😊

---

I have a meta question: When I force-push in a PR (which y'all do all the time), you can't make out what I last changed, right? If the PR contained many changes, and the force-push doesn't throw everything over, but only changes a little thing, is a force push annoying to you and adding a small commit preferred?

If yes, should these changes, if I ideally would have presented them from the get-go in a single commit anyways (or a well-defined series of commits) at the end still be consolidated with a final force push? That would require you to trust me, though, that I didn't change something else also, right?

**Comment by tronical** [comment]
> > I'll still hold off until after the release
> 
> Ah, so not until right before the release of v1.9. Then, this is not so bad. 😊

Right, I'll merge this right after the release of 1.8.

> I have a meta question: When I force-push in a PR (which y'all do all the time), you can't make out what I last changed, right?

That's correct.

>  If the PR contained many changes, and the force-push doesn't throw everything over, but only changes a little thing, is a force push annoying to you and adding a small commit preferred?

I think it depends on the overall size of the patch. The larger it is, the longer it takes to re-review it as a whole, which typically happens after a force push. You can always add small commits on top to make the review process easier, and then squash before merging.

> If yes, should these changes, if I ideally would have presented them from the get-go in a single commit anyways (or a well-defined series of commits) at the end still be consolidated with a final force push? 

That's indeed a method we're using occasionally. That said, concretely on this PR, I think it's fine to force-push as it's a largely mechanical change that's easier to review. Suppose you're writing a complex algorithm and during review you'd like to change some parts of it, then a series is easier to review.

> That would require you to trust me, though, that I didn't change something else also, right?

That's correct - trust plays an important role here :)

**Comment by tronical** [comment]
Olivier suggested to me to still put this into 1.8, so I'll go ahead and rebase & edit this PR to accommodate. So don't get scared if I push into your branch :)

**Pull request merged**